src/app/chat/views/messages.nim

```diff
@@ -100,11 +57,34 @@ QtObject:
     result.messageList[status_utils.getTimelineChatId()] = newChatMessageList(status_utils.getTimelineChatId(), result.status, false)
     result.loadingMessages = false
     result.unreadMessageCnt = 0
-    result.observedMessageItem = newMessageItem(status)
+    result.searchResultMessageModel = newMessageListProxyModel(status)
     result.unreadDirectMessagesAndMentionsCount = 0
+    result.loadingHistoryMessages = false
     result.setup
 
-  # proc getMessageListIndexById(self: MessageView, id: string): int
+  #################################################
+  # Forward declaration section
+  #################################################
```

Comment:
I am not sure if there is consensus here on those type of comment but if there isn't that will quickly become unmaintained and source of confusion

Reply:
@alaibe we should agreed on this, if that's an issue I can remove it, it's not a problem at all, but I found this kind of headers like "Forward declaration section", "Property section" very useful cause that increases readability and avoids some duplications I would say,

Reply:
I don't have personal preference on that, for or against, I am happy to follow. So for me not an issue but as soon as we all follow this guideline I am good!

Reply:
@jrainville when I say "Properties" I referred to QtProperty. Yes, they are simple functions, but generally speaking they are usually short and do a single action (it's not recommended to have WRITE property which will do a lot of other calls) they are created to update a view (qml) quickly. 

In c++ you may use Q_PROPERTY macro in header files and when you look the class header all properties are on the same place and you may get an easy idea what was an intention there and what is exposed to qml as property. We don't have header files here and just wanted to wanted to emphasize part where we're stating Qt's properties, that's just a matter of style (I don't like to search through the whole file for a properties), but as you decide.

Reply:
@jrainville when I say "Properties" I referred to QtProperty. Yes, they are simple functions, but generally speaking they are usually short and do a single action (it's not recommended to have WRITE property which will do a lot of other calls) they are created to update a view (qml) quickly.

In c++ you may use Q_PROPERTY macro in header files and when you look the class header all properties are on the same place and you may get an easy idea what was an intention there and what is exposed to qml as property. We don't have header files here and just wanted to wanted to emphasize part where we're stating Qt's properties, that's just a matter of style (I don't like to search through the whole file for a properties), but as you decide.